submissions/ses3332002/dom/script.js

```diff
@@ -0,0 +1,257 @@
+      "Jam",
+      "Why You Wanna Trip on Me",
+      "In the Closet",
+      "She Drives Me Wild",
+      "Remember the Time",
+      "Can't Let Her Get Away",
+      "Heal the World",
+      "Black or White",
+      "Who Is It",
+      "Give in to Me",
+      "Will You Be There",
+      "Keep the Faith",
+      "Gone Too Soon",
+      "Dangerous"
+    ],
+    "bgColor": 29
+  },
+  {
+    "title": "HIStory: Past, Present and Future, Book I (1995)",
+    "cover": "history",
+    "songs": [
+      "Scream",
+      "They Don’t Care About Us",
+      "Stranger in Moscow",
+      "This Time Around",
+      "Earth Song",
+      "D.S.",
+      "Money",
+      "Come Together",
+      "You Are Not Alone",
+      "Childhood",
+      "Tabloid Junkie",
+      "2 Bad",
+      "HIStory",
+      "Little Susie",
+      "Smile"
+    ],
+    "bgColor": 332
+  },
+  {
+    "title": "Invincible (2001)",
+    "cover": "invincible",
+    "songs": [
+      "Unbreakable",
+      "Heartbreaker",
+      "Invincible",
+      "Break of Dawn",
+      "Heaven Can Wait",
+      "You Rock My World",
+      "Butterflies",
+      "Speechless",
+      "2000 Watts",
+      "You Are My Life",
+      "Privacy",
+      "Don't Walk Away",
+      "Cry",
+      "The Lost Children",
+      "Whatever Happens",
+      "Threatened"
+    ],
+    "bgColor": 215
+  },
+  {
+    "title": "Michael (2010)",
+    "cover": "michael",
+    "songs": [
+      "Hold My Hand",
+      "Hollywood Tonight",
+      "Keep Your Head Up",
+      "(I Like) The Way You Love Me",
+      "Monster",
+      "Best of Joy",
+      "Breaking News",
+      "(I Can't Make It) Another Day",
+      "Behind the Mask",
+      "Much Too Soon"
+    ],
+    "bgColor": 8
+  },
+  {
+    "title": "Xscape (2014)",
+    "cover": "xscape",
+    "songs": [
+      "Love Never Felt So Good",
+      "Chicago",
+      "Loving You",
+      "A Place with No Name",
+      "Slave to the Rhythm",
+      "Do You Know Where Your Children Are",
+      "Blue Gangsta",
+      "Xscape"
+    ],
+    "bgColor": 33
+  }
+];
+
+let doc = document.body;
+let navTab = document.querySelector(".navigation");
+let infoTab = document.querySelector(".info");
+let navTabButtonsArray;
+
+function fillNavTab() {
+  navTab.innerHTML = "";
+  let navTabButton;
+  for (let i = 0; i < albums.length; i++) {
+    navTabButton = document.createElement("button");
+    navTabButton.classList.add("navigation_button");
+    navTabButton.innerText = albums[i].title;
+    navTab.append(navTabButton);
+  };
+  navTabButtonsArray = Array.from(document.querySelectorAll(".navigation_button"));
+};
+
+function fillInfoTab(e) {
+  let selectedButtonIndex = navTabButtonsArray.indexOf(e.target);
+  let infoList;
+  let infoLine;
+  infoTab.innerHTML = "";
+  doc.style.setProperty("--background_color_h", albums[selectedButtonIndex].bgColor);
+  infoTab.style.backgroundImage = "url(img/" + albums[selectedButtonIndex].cover + ".jpg)";
+  infoList = document.createElement("ul");
```

Comment:
You can move this `ul` to the HTML and generate `li` elements here. Also, I recommend using `DocumentFragment` here

Reply:
:) I use "ul" here exactly for making only one DOM insertion.
OK, I can remake this with DocumentFragment.

Reply:
Ready. I removed "ul" to HTML and used DocumentFragment to insert in DOM only once.
But as a result my HTML became bigger, my CSS became bigger ('cos I needed to style both "ul" and "li") and my JS became bigger. Honestly: I hardly understand such improvements :)

Reply:
1. HTML became bigger per one element? 
2. Why did your CSS become bigger? How did changes in JS affect your styles?
3. Your code became more clear and easy to read.

Reply:
1. Yes, HTML became bigger per one element, which is absolutely not needed in "initial" screen (while no button pressed).
2. Because of 1 I needed to remake styles - make this "ul" clear and transparent and give background and paddings to "li" (and reset ugly defaults for both of them).
3. Ok. But I'm afraid that some third reviewer could find another problems, like unnessessery variables, used only for DOM manipulations :)

Reply:
@ses3332002 Look, your decision was totally fine. And I appreciate all work you've done! Your code was nice and it worked perfectly.
My idea was to move all elements to HTML (if the situation allows) and to decrease DOM operations in your javascript because as we know, DOM operations cost the most. Maybe, it caused some changes in CSS. But making changes is CSS is no big deal, to my mind. 

BTW, I don't urge you to follow all of my comments and make these changes. There are just recommendations that I learned during my work experience. You can't agree with all of them. It's ok. It's totally up to you. I open to discussions if we need it.

Reply:
It's Ok, I'm just grumble :)